### PR TITLE
OCPBUGS-46483: Correctly Reconcile CSO CSI Secrets for Managed Azure Deployments

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
This PR:
- removes the double reconciliation of the CSO CSI secrets for managed azure deployments. Previous to the managed identity work merging into the HyperShift repo, we were reconciling the CSO CSI secrets with the Cloud Provider information. Once the managed identity work was added, the reconciliation with the Cloud Provider information wasn't removed.
- adds the file path with the certificate name for the CSO CSI secret reconciliation for managed azure deployments.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes ssue(s) this PR fixes :
Fixes # [OCPBUGS-46483](https://issues.redhat.com/browse/OCPBUGS-46483)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/storage/azure.go [hunk 1]
```diff
@@ -28,7 +29,7 @@ func initializeAzureCSIControllerConfig(hcp *hyperv1.HostedControlPlane, tenantI
 func ReconcileAzureDiskCSISecret(secret *corev1.Secret, hcp *hyperv1.HostedControlPlane, tenantID string) error {
 	config := initializeAzureCSIControllerConfig(hcp, tenantID)
 	config.AADClientID = hcp.Spec.Platform.Azure.ManagedIdentities.ControlPlane.Disk.ClientID
-	config.AADClientCertPath = hcp.Spec.Platform.Azure.ManagedIdentities.ControlPlane.Disk.CertificateName
+	config.AADClientCertPath = hypershiftconfig.ManagedAzureCertificatePath + hcp.Spec.Platform.Azure.ManagedIdentities.ControlPlane.Disk.CertificateName
```
Comment (muraee) [inline control-plane-operator/controllers/hostedcontrolplane/storage/azure.go:32]: please use `path.Join()` here instead of `+`

**Comment by bryan-cox** [comment]
/hold 

There is some issue with one of the secrets

**Comment by bryan-cox** [comment]
/unhold

Fixed secret problem

**Comment by bryan-cox** [comment]
/retest

For konflux

**Comment by bryan-cox** [comment]
/override "Red Hat Konflux / hypershift-operator-main-on-pull-request"

**Pull request merged**